src/Microsoft.ML.Data/Transforms/KeyToVectorTransform.cs

```diff
@@ -69,530 +75,776 @@ public bool TryUnparse(StringBuilder sb)
+            public readonly bool Bag;
+            public Config(bool bag)
+            {
+                Bag = bag;
+            }
+        }
 
-                    getSrc(ref src);
-                    Host.Check(cv == 0 || src.Length == cv);
+        private interface IColInput
+        {
+            PipelineColumn Input { get; }
+            Config Config { get; }
+        }
 
-                    // The indices are irrelevant in the bagging case.
-                    var values = src.Values;
-                    int count = src.Count;
-                    for (int slot = 0; slot < count; slot++)
-                    {
-                        uint key = values[slot] - 1;
-                        if (key < size)
-                            bldr.AddFeature((int)key, 1);
-                    }
+        private sealed class OutVectorColumn<TKey, TValue> : Vector<float>, IColInput
+        {
+            public PipelineColumn Input { get; }
+            public Config Config { get; }
 
-                    bldr.GetResult(ref dst);
-                };
+            public OutVectorColumn(Vector<Key<TKey, TValue>> input, Config config)
+                : base(Reconciler.Inst, input)
+            {
+                Input = input;
+                Config = config;
+            }
+
+            public OutVectorColumn(Key<TKey, TValue> input, Config config)
+              : base(Reconciler.Inst, input)
+            {
+                Input = input;
+                Config = config;
+            }
+        }
+
+        private sealed class Reconciler : EstimatorReconciler
+        {
+            public static Reconciler Inst = new Reconciler();
+
+            private Reconciler() { }
+
+            public override IEstimator<ITransformer> Reconcile(IHostEnvironment env,
+                PipelineColumn[] toOutput,
+                IReadOnlyDictionary<PipelineColumn, string> inputNames,
+                IReadOnlyDictionary<PipelineColumn, string> outputNames,
+                IReadOnlyCollection<string> usedNames)
+            {
+                var infos = new KeyToVectorTransform.ColumnInfo[toOutput.Length];
+                for (int i = 0; i < toOutput.Length; ++i)
+                {
+                    var col = (IColInput)toOutput[i];
+                    infos[i] = new KeyToVectorTransform.ColumnInfo(inputNames[col.Input], outputNames[toOutput[i]], col.Config.Bag);
+                }
+                return new KeyToVectorEstimator(env, infos);
+            }
         }
 
         /// <summary>
-        /// This is for the indicator (non-bagging) case - vector input and outputs should be concatenated.
+        /// Takes a column of key type of known cardinality and produces an indicator vector of floats.
         /// </summary>
-        private ValueGetter<VBuffer<Float>> MakeGetterInd(IRow input, int iinfo)
+        public static Vector<float> ToVector<TKey, TValue>(this Key<TKey, TValue> input, bool bag = DefaultBag)
```

Comment:
>Key<TKey, TValue> [](start = 64, length = 17)

Hi @ivanidzo4ka. I suggest you change the input type from `Key<TKey, TValue>` to just plain old `Key<T>`. Whether or not there is key-value metadata is immaterial to the functioning of this transform, and `Key<T,V>` descends from `Key<T>` anyway.

Reply:
is this comment only for this case, or in general across  all methods?

---
In reply to: [216801604](https://github.com/dotnet/machinelearning/pull/858#discussion_r216801604) [](ancestors = 216801604)

Reply:
Everything ought to allow a value of `Key<T>` and `Key<T,V>`, by some mechanism. I had, however, forgotten about `Vector<Key<T,V>>` not being a `Vector<Key<T>>`... this will require a separate overload I think.

---
In reply to: [216812660](https://github.com/dotnet/machinelearning/pull/858#discussion_r216812660) [](ancestors = 216812660,216801604)

Reply:
Ok, I added Key<TKey> and Key<TKey,TValue> extensions

---
In reply to: [216841257](https://github.com/dotnet/machinelearning/pull/858#discussion_r216841257) [](ancestors = 216841257,216812660,216801604)